### PR TITLE
Support dlsproto v14.1.0 (including neo support)

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
This PR adds support for neo requests in the official (master) build of dlsnode.

### CODE REVIEW

**Comment by nemanja-boric-sociomantic** [comment]
Blocked due to dlsproto v14.1.0 release.

Has to be merged manually, as it merges `neo` into `master`.

**Comment by nemanja-boric-sociomantic** [comment]
Squashed WIP commits and set the order of the commits for the merge commit to come first. Unblocked.

File: src/dlsnode/app/config/PerformanceConfig.d [hunk 1]
```diff
@@ -49,5 +49,15 @@ public class PerformanceConfig
...
+
+    /***************************************************************************
+
+        Disable Nagle's algorithm for neo sockets. This is important for
+        testing requests with large volumes of data -- Nagle's algorithm slows
+        them down too much, if left enabled.
+
+    ***************************************************************************/
+
+    bool no_delay = false;
```
Comment (gavin-norman-sociomantic) [inline src/dlsnode/app/config/PerformanceConfig.d]: Wait, doesn't `no_delay = false` mean "Nagle's algorithm on"? (Mismatch between comment and code.)

File: src/dlsnode/storage/protocol/StorageProtocolLegacy.d [hunk 2]
```diff
@@ -15,9 +15,12 @@ module dlsnode.storage.protocol.StorageProtocolLegacy;
 import ocean.transition;
 
 import dlsnode.storage.protocol.model.IStorageProtocol;
+import dlsnode.storage.util.Promise;
```
Comment (gavin-norman-sociomantic) [inline src/dlsnode/storage/protocol/StorageProtocolLegacy.d]: Do we still need the legacy storage protocol btw? (Could be removed in a separate PR.)

File: src/dlsnode/connection/neo/SharedResources.d [hunk 3]
```diff
@@ -346,5 +346,10 @@ public final class SharedResources
         {
             return this.acquired_regexes.acquire(this.outer.pcre.new CompiledRegex);
         }
+
+        public StorageChannels storage_channels ()
```
Comment (gavin-norman-sociomantic) [inline src/dlsnode/connection/neo/SharedResources.d]: Missing doc.

File: src/dlsnode/app/config/PerformanceConfig.d (hunk 1 above)
Comment (nemanja-boric-sociomantic) [reply src/dlsnode/app/config/PerformanceConfig.d]: Right - isn't that what is says: "Disable Nagle's algorithm for neo sockets"?

File: src/dlsnode/storage/protocol/StorageProtocolLegacy.d (hunk 2 above)
Comment (nemanja-boric-sociomantic) [reply src/dlsnode/storage/protocol/StorageProtocolLegacy.d]: Right, it's no longer needed, I'll do it in the next PR.

File: src/dlsnode/connection/neo/SharedResources.d (hunk 3 above)
Comment (nemanja-boric-sociomantic) [reply src/dlsnode/connection/neo/SharedResources.d]: Updated.

File: src/dlsnode/storage/protocol/StorageProtocolLegacy.d (hunk 2 above)
Comment (nemanja-boric-sociomantic) [reply src/dlsnode/storage/protocol/StorageProtocolLegacy.d]: https://github.com/sociomantic-tsunami/dlsnode/issues/50

File: src/dlsnode/app/config/PerformanceConfig.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/dlsnode/app/config/PerformanceConfig.d]: No:
* Nagle's algorithm disabled == `no_delay = true`.
* Nagle's algorithm enabled == `no_delay = false`.

(Or am I going crazy?)

Comment (nemanja-boric-sociomantic) [reply src/dlsnode/app/config/PerformanceConfig.d]: You're not going crazy, perhaps I'm missing the point :-). When one sets `no_delay` to `true` we do what the documentation says: `Disable Nagle's algorithm...`, right?

I'll just remove this field ;-).

Comment (nemanja-boric-sociomantic) [reply src/dlsnode/app/config/PerformanceConfig.d]: There ;-).
